lib/kitchen/directions/bake_chapter_key_concepts.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Kitchen
+  module Directions
+    # Bake directions for eoc key concepts
+    #
+    module BakeChapterKeyConcepts
+      def self.v1(chapter:, metadata_source:)
+        metadata_elements = metadata_source.children_to_keep.copy
+
+        fix_titles = chapter.pages.map do |page|
+          key_concepts = page.search('section.key-concepts')
+          next if key_concepts.none?
+
+          key_concepts.search('h3').trash
+          key_concepts.map do |key_concept|
+            id = key_concept.id.split('fs-id')
+            key_concept.prepend(child:
+              <<~HTML
+                <a href="##{id[0]}0">
+                  <h3 data-type="document-title" id="#{id[0]}0_copy_XXX">
+                    <span class="os-number">#{chapter.count_in(:book)}.#{key_concept.count_in(:chapter)}</span>
+                    <span class="os-divider"> </span>
+                    <span class="os-text" data-type="" itemprop="">#{page.title.text}</span>
+                  </h3>
+                </a>
+              HTML
+            )
+
+            <<~HTML
+              <div class="os-section-area">
+                #{key_concept}
+              </div>
+            HTML
+          end.compact.join("\n")
```

Comment:
Do you need the `compact`?  Would only be useful if `key_concept` were nil (`compact` gets rid of `nil` values in arrays).  Ditto for next line.